src/api/api.ts

```diff
@@ -151,7 +158,29 @@ export const logInUser = async ({
     email: resultUserInfo.user.email!,
     id: resultUserInfo.user.uid,
     photoURL: resultUserInfo.user.photoURL!,
+    createdAt: resultUserInfo.user.metadata.creationTime,
+    fullName: resultUserInfo.user.displayName,
+    phone: resultUserInfo.user.phoneNumber,
   };
 
   return { isUser, resultUserInfoData };
 };
+
+export const setDataStorage = async ({
+  fileCollection,
+  file,
+}: IUploadDataStorage): Promise<UploadTaskSnapshot> => {
+  const imagesRef = ref(storage, `${fileCollection}/${file.name}`);
+
+  const uploadTask = await uploadBytesResumable(imagesRef, file);
+
+  return uploadTask;
+};
+
+export const getDataStorage = async (fileName: string): Promise<string> => {
+  const imagesRef = ref(storage, `images/${fileName}`);
+
+  const imageUrl = await getDownloadURL(imagesRef);
+
+  return imageUrl;
+};
```

Comment:
I don't think we need to merge the logic here, since we don't always need a link to the file being uploaded, and making additional unnecessary requests = overloading the system.

Reply:
Whenever you load an image, you will immediately need to save a link to it in the database, so I recommend that you return the URL of the image